File: check_braces.js
const fs = require('fs');
const content = fs.readFileSync('app/page.tsx', 'utf8');

function checkBalance(text) {
    let parens = 0;
    let braces = 0;
    let insideString = null;
    let i = 0;
    while (i < text.length) {
        const char = text[i];
        const nextChar = text[i+1];

        if (insideString) {
            if (char === '\\') { i += 2; continue; }
            if (char === insideString) { insideString = null; }
        } else {
            if (char === '"' || char === "'" || char === '`') { insideString = char; }
            else if (char === '/' && nextChar === '/') { i = text.indexOf('\n', i); if (i === -1) break; }
            else if (char === '/' && nextChar === '*') { i = text.indexOf('*/', i); if (i === -1) break; i += 1; }
            else if (char === '(') parens++;
            else if (char === ')') parens--;
            else if (char === '{') braces++;
            else if (char === '}') braces--;
        }
        i++;
    }
    console.log(`Balance - Parens: ${parens}, Braces: ${braces}`);
}

checkBalance(content);

File: check_braces.py

import sys

def check_balance(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()
    
    stack = []
    lines = content.split('\n')
    for i, line in enumerate(lines):
        line_num = i + 1
        for char in line:
            if char == '{':
                stack.append(('{', line_num))
            elif char == '}':
                if not stack:
                    print(f"Extra '}}' at line {line_num}")
                    return
                stack.pop()
    
    if stack:
        for char, line_num in stack:
            print(f"Unclosed '{char}' from line {line_num}")
    else:
        print("Braces are balanced")

if __name__ == "__main__":
    check_balance(r"c:\Hafiz Wrg\database\Game\app\page.tsx")
